mixer/adapter/stackdriver/contextgraph/workload.go

```diff
@@ -197,10 +237,32 @@ func (t trafficAssertion) Reify(logger adapter.Logger) ([]entity, []edge) {
 				edges = append(edges, edge{s, d, typeName})
 			}
 		}
+		edges = append(edges, edge{srcInst, serviceEntity.fullName, typeName})
+		edges = append(edges, edge{srcOwn, serviceEntity.fullName, typeName})
+		edges = append(edges, edge{serviceEntity.fullName, dstInst, typeName})
+		edges = append(edges, edge{serviceEntity.fullName, dstOwn, typeName})
```

Comment:
I think we can do away with saving the local vars above, and simply loop through the `sourceFullNames` and add an edge from each to the service, then do the same from the service to each name in `destinationFullNames`

Reply:
ok. i refactored a bit to cut down on the looping and local vars. I've added a unit test to cover the code that probably needs a good look over to verify the correct results are being produced. I'll take a linting pass next.